### PR TITLE
Add groups to Security chapter's TOC

### PR DESCRIPTION
**Pull request by Ordinant**
This is stage 1 of the security rework proposed for Trino docs.

### CODE REVIEW

**Comment by Ordinant** [comment]
Attached image is what the Security section's TOC will look like.
Proposed_Trino_Security_TOC

File: docs/src/main/sphinx/security.rst [hunk 1]
```diff
@@ -2,18 +2,41 @@
...
+
+.. toctree::
+  :maxdepth: 1
+
+  security/tls
+  security/ldap
+  security/password-file
+  security/salesforce
+
+User name management
```
Comment (mosabua) [inline docs/src/main/sphinx/security.rst:16]: User management ?

**Review by mosabua — APPROVED** [review]
We could go with Authentication and Authorization but I prefer the current setup since in my experience most people mixed the two up..

File: docs/src/main/sphinx/security.rst (hunk 1 above)
Comment (Ordinant) [reply docs/src/main/sphinx/security.rst:16]: Could go with that. I was just trying to match what SEP docs have now.

**Review by electrum — APPROVED** [review]
@dain do these groupings and titles look good to you?

File: docs/src/main/sphinx/security.rst (hunk 1 above)
Comment (dain) [reply docs/src/main/sphinx/security.rst:16]: This is likely fine for now, but as we add more group providers, we will likely need to reorganize

**Review by dain — APPROVED** [review]
Looks good

**Comment by mosabua** [comment]
Thats good. Can one of you please merge @dain or @electrum ?

**Comment by mosabua** [comment]
Alternatively any other maintainer ready to merge?  @trinodb/maintainers

**Pull request merged**